### ISSUE TITLE
Working on vhost, not working on apparently identical vhost

### ISSUE THREAD

**Issue by rr4444** [nginx-proxy/docker-letsencrypt-nginx-proxy-companion#659]
Hi there,

I am going mad.... nginx-proxy is installed and working on first container below (pasted key parts from docker inspect). But the second is basically identical and not working. It was working until this morning, I didn't do anything to it that I recall... But I was messing around with the nginx/letsenc setup. I have killed the nginx/letsenc setup and its volumes and re run those two containers several times just in case. 

I get a 500 error on that second backend container when it is awake (503 when down, which is expected i guess). The 500 error is apparently a cert problem because the cert req is failing with:

> CA marked some of the authorizations as invalid

for which I have disabled ipv6, which didnt seem to be needed in this case, but I am getting frustrated and trying anything! All the domains are on cloudflare, so it's not the DNS provider. It doesn't even appear to arrive on the backend container httpd, cos nginx is falling over first at https. 

If I force a renew, the working one renews happily. And the others fail in the same way. 

I have directly tested whether https is responding via the exposed 8443 port on the IP set via the nginx-proxy docker network subnet, which is set for the VIRTUAL_PORTs defined in ENV vars. https is responding happily on the exposed (unpublished) ports. Going mad...

If I run the below, it confirms the same behaviour, where i get Hello World on the working vhost, but a 500 error on the other:

`you@remotedockerhost$ docker exec your-le-container bash -c 'echo "Hello world!" > /usr/share/nginx/html/.well-known/acme-challenge/hello-world'`

`you@localcomputer$ curl http://yourdomain.tld/.well-known/acme-challenge/hello-world
Hello world!`
`you@localcomputer$ curl -6 http://yourdomain.tld/.well-known/acme-challenge/hello-world
Hello world!`

**Some of inspect contents from the containers (sensitive details removed inc. domains)**

Working one:

` "Env": [
            "APACHE_HTTPS_PORT_NUMBER=8443",
            "APACHE_HTTP_PORT_NUMBER=8080",
            "WORDPRESS_HTTPS_PORT=8443",
            "WORDPRESS_HTTP_PORT=8080",
            "VIRTUAL_HOST=dev.em.com",
            "VIRTUAL_PORT=8443",
            "VIRTUAL_PROTO=https",
            "LETSENCRYPT_HOST=dev.em.com",
        ],
        "ExposedPorts": {
            "443/tcp": {},
            "8080/tcp": {},
            "8443/tcp": {}
        },`

Not working:

`  "Env": [
            "APACHE_HTTPS_PORT_NUMBER=8443",
            "APACHE_HTTP_PORT_NUMBER=8080",
            "WORDPRESS_HTTPS_PORT=8443",
            "WORDPRESS_HTTP_PORT=8080",
            "VIRTUAL_HOST=lux.rr.com",
            "VIRTUAL_PORT=8443",
            "VIRTUAL_PROTO=https",
            "LETSENCRYPT_HOST=lux.rr.com",
        ],
        "ExposedPorts": {
            "443/tcp": {},
            "8080/tcp": {},
            "8443/tcp": {}
        },`

**Comment by rr4444** [comment]
Sharing for completeness, but I think this is fine, or the first working backend container wouldn't work at all:

`docker run --detach  --name nginx-proxy  --publish 80:80 --publish 443:443    --volume /etc/nginx/certs    --volume /etc/nginx/vhost.d   --volume /usr/share/nginx/html    --volume /var/run/docker.sock:/tmp/docker.sock:ro --network nginx-proxy  jwilder/nginx-proxy:alpine`

`docker run --detach  --name nginx-letsencrypt  --volumes-from nginx-proxy  --volume /var/run/docker.sock:/var/run/docker.sock:ro    --env "DEFAULT_EMAIL=r@notmyemail.com"   jrcs/letsencrypt-nginx-proxy-companion`

**Comment by rr4444** [comment]
My working assumption is that the failure to get the certs means that this is a problem with letsenc side of things breaking the nginx response i.e. and not the nginx-proxy itself. 

Unless I am wrong?

**Comment by rr4444** [comment]
There is a third container, not identical, but that is not even started. i am keeping that off for now...

**Comment by rr4444** [comment]
Many thanks in advance for any other pointers or ideas!!!!!!!

I looked at traefik, but this setup is a better simpler approach for little old me. Thanks again.

**Comment by buchdag** [comment]
> The 500 error is apparently a cert problem because the cert req is failing with: CA marked some of the authorizations as invalid

My guess is that it's probably the other way around, you can't get a certificate because of the proxying error.

You should have a link providing some additional infos about why the challenge failed on the `letsencrypt-nginx-proxy-companion` container logs.

> `APACHE_HTTPS_PORT_NUMBER`
> `APACHE_HTTP_PORT_NUMBER`
> `WORDPRESS_HTTPS_PORT`
> `WORDPRESS_HTTP_PORT`

What are those environment variables ?

Could you post the rendered `/etc/nginx/conf.d/default.conf` file when both containers are up ?

**Comment by rr4444** [comment]
Sorry for the delayed reply - the joys of pandemic :(

Firstly, these are the verbose results of forcing a renew, with only the first container working, like before (there are 2 WP containers then a Joomla):

> root@docker-s-1vcpu-2gb-lon1-01:~# docker exec nginx-letsencrypt /app/force_renew -v                
> /etc/nginx/certs/dev.2.com /app
> Creating/renewal dev.2.com certificates... (dev.2.com)
> 2020-05-27 17:00:51,829:INFO:simp_le:1450: Generating new certificate private key
> 2020-05-27 17:00:55,904:INFO:simp_le:401: Saving key.pem
> 2020-05-27 17:00:55,905:INFO:simp_le:401: Saving fullchain.pem
> 2020-05-27 17:00:55,906:INFO:simp_le:401: Saving cert.pem
> /app
> /etc/nginx/certs/dev.1.com /app
> Creating/renewal dev.1.com certificates... (dev.1.com)
> 2020-05-27 17:00:57,476:INFO:simp_le:1450: Generating new certificate private key
> 2020-05-27 17:01:03,832:ERROR:simp_le:1417: CA marked some of the authorizations as invalid, which likely means it could not access http://example.com/.well-known/acme-challenge/X. Did you set correct path in -d example.com:path or --default_root? Are all your domains accessible from the internet? Please check your domains' DNS entries, your host's network/firewall setup and your webserver config. If a domain's DNS entry has both A and AAAA fields set up, some CAs such as Let's Encrypt will perform the challenge validation over IPv6. If your DNS provider does not answer correctly to CAA records request, Let's Encrypt won't issue a certificate for your domain (see https://letsencrypt.org/docs/caa/). Failing authorizations: https://acme-v02.api.letsencrypt.org/acme/authz-v3/4848418835
> Challenge validation has failed, see error log.
> 
> Debugging tips: -v improves output verbosity. Help is available under --help.
> /app
> /etc/nginx/certs/lux.1.com /app
> Creating/renewal lux.1.com certificates... (lux.1.com)
> 2020-05-27 17:01:05,247:INFO:simp_le:1450: Generating new certificate private key
> 2020-05-27 17:01:09,184:ERROR:simp_le:1417: CA marked some of the authorizations as invalid, which likely means it could not access http://example.com/.well-known/acme-challenge/X. Did you set correct path in -d example.com:path or --default_root? Are all your domains accessible from the internet? Please check your domains' DNS entries, your host's network/firewall setup and your webserver config. If a domain's DNS entry has both A and AAAA fields set up, some CAs such as Let's Encrypt will perform the challenge validation over IPv6. If your DNS provider does not answer correctly to CAA records request, Let's Encrypt won't issue a certificate for your domain (see https://letsencrypt.org/docs/caa/). Failing authorizations: https://acme-v02.api.letsencrypt.org/acme/authz-v3/4848421060
> Challenge validation has failed, see error log.
> 
> Debugging tips: -v improves output verbosity. Help is available under --help.
> /app
> Reloading nginx proxy (nginx-proxy)...
> 2020/05/27 17:01:10 Generated '/etc/nginx/conf.d/default.conf' from 9 containers
> 2020/05/27 17:01:10 [notice] 285#285: signal process started
> root@docker-s-1vcpu-2gb-lon1-01:~# 
> 

Environment vars on the first two vhosts in the logs, both WP containers and only the first works:

> APACHE_HTTPS_PORT_NUMBER | 8443
> -- | --
> APACHE_HTTP_PORT_NUMBER | 8080
> 
> WORDPRESS_HTTPS_PORT | 8443
> -- | --
> WORDPRESS_HTTP_PORT | 8080
> 
> 
The last one is an official joomla container happily answering directly via http on port 80 with no problems, but not via nginx either. 

This is the nginx.conf

> bash-5.0# more /etc/nginx/conf.d/default.conf
> # If we receive X-Forwarded-Proto, pass it through; otherwise, pass along the
> # scheme used to connect to this server
> map $http_x_forwarded_proto $proxy_x_forwarded_proto {
>   default $http_x_forwarded_proto;
>   ''      $scheme;
> }
> # If we receive X-Forwarded-Port, pass it through; otherwise, pass along the
> # server port the client connected to
> map $http_x_forwarded_port $proxy_x_forwarded_port {
>   default $http_x_forwarded_port;
>   ''      $server_port;
> }
> # If we receive Upgrade, set Connection to "upgrade"; otherwise, delete any
> # Connection header that may have been passed to this server
> map $http_upgrade $proxy_connection {
>   default upgrade;
>   '' close;
> }
> # Apply fix for very long server names
> server_names_hash_bucket_size 128;
> # Default dhparam
> ssl_dhparam /etc/nginx/dhparam/dhparam.pem;
> # Set appropriate X-Forwarded-Ssl header
> map $scheme $proxy_x_forwarded_ssl {
>   default off;
>   https on;
> }
> gzip_types text/plain text/css application/javascript application/json application/x-javascript text/xml application/xml application/xml+rss text/javascript;
> log_format vhost '$host $remote_addr - $remote_user [$time_local] '
>                  '"$request" $status $body_bytes_sent '
>                  '"$http_referer" "$http_user_agent"';
> access_log off;
>                 ssl_protocols TLSv1.2 TLSv1.3;
>                 ssl_ciphers 'ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384';
>                 ssl_prefer_server_ciphers off;
> resolver 127.0.0.11;
> # HTTP 1.1 support
> proxy_http_version 1.1;
> proxy_buffering off;
> proxy_set_header Host $http_host;
> proxy_set_header Upgrade $http_upgrade;
> proxy_set_header Connection $proxy_connection;
> proxy_set_header X-Real-IP $remote_addr;
> proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
> proxy_set_header X-Forwarded-Proto $proxy_x_forwarded_proto;
> proxy_set_header X-Forwarded-Ssl $proxy_x_forwarded_ssl;
> proxy_set_header X-Forwarded-Port $proxy_x_forwarded_port;
> # Mitigate httpoxy attack (see README for details)
> proxy_set_header Proxy "";
> server {
>         server_name _; # This is just an invalid value which will never trigger on a real hostname.
>         listen 80;
>         access_log /var/log/nginx/access.log vhost;
>         return 503;
> }
> server {
>         server_name _; # This is just an invalid value which will never trigger on a real hostname.
>         listen 443 ssl http2;
>         access_log /var/log/nginx/access.log vhost;
>         return 503;
>         ssl_session_cache shared:SSL:50m;
>         ssl_session_tickets off;
>         ssl_certificate /etc/nginx/certs/default.crt;
>         ssl_certificate_key /etc/nginx/certs/default.key;
> }
> # dev.2.com
> upstream dev.2.com {
>                                 ## Can be connected with "nginx-proxy" network
>                         # wp_evemeetsadam
>                         server 172.18.0.3:8443;
>                                 # Cannot connect to network of this container
>                                 server 127.0.0.1 down;
> }
> server {
>         server_name dev.2.com;
>         listen 80 ;
>         access_log /var/log/nginx/access.log vhost;
>         # Do not HTTPS redirect Let'sEncrypt ACME challenge
>         location /.well-known/acme-challenge/ {
>                 auth_basic off;
>                 allow all;
>                 root /usr/share/nginx/html;
>                 try_files $uri =404;
>                 break;
>         }
>         location / {
>                 return 301 https://$host$request_uri;
>         }
> }
> server {
>         server_name dev.2.com;
>         listen 443 ssl http2 ;
>         access_log /var/log/nginx/access.log vhost;
>         ssl_session_timeout 5m;
>         ssl_session_cache shared:SSL:50m;
>         ssl_session_tickets off;
>         ssl_certificate /etc/nginx/certs/dev.2.com.crt;
>         ssl_certificate_key /etc/nginx/certs/dev.2.com.key;
>         ssl_dhparam /etc/nginx/certs/dev.2.com.dhparam.pem;
>         ssl_stapling on;
>         ssl_stapling_verify on;
>         ssl_trusted_certificate /etc/nginx/certs/dev.2.com.chain.pem;
>         add_header Strict-Transport-Security "max-age=31536000" always;
>         include /etc/nginx/vhost.d/default;
>         location / {
>                 proxy_pass https://dev.2.com;
>         }
> }
> # dev.1.com
> upstream dev.1.com {
>                                 # Cannot connect to network of this container
>                                 server 127.0.0.1 down;
>                                 ## Can be connected with "nginx-proxy" network
>                         # joomla_RRcom
>                         server 172.18.0.5:80;
> }
> server {
>         server_name dev.1.com;
>         listen 80 ;
>         access_log /var/log/nginx/access.log vhost;
>         include /etc/nginx/vhost.d/default;
>         location / {
>                 proxy_pass http://dev.1.com;
>         }
> }
> server {
>         server_name dev.1.com;
>         listen 443 ssl http2 ;
>         access_log /var/log/nginx/access.log vhost;
>         return 500;
>         ssl_certificate /etc/nginx/certs/default.crt;
>         ssl_certificate_key /etc/nginx/certs/default.key;
> }
> # lux.1.com
> upstream lux.1.com {
>                                 # Cannot connect to network of this container
>                                 server 127.0.0.1 down;
>                                 ## Can be connected with "nginx-proxy" network
>                         # wp_lux
>                         server 172.18.0.4:8443;
> }
> server {
>         server_name lux.1.com;
>         listen 80 ;
>         access_log /var/log/nginx/access.log vhost;
>         include /etc/nginx/vhost.d/default;
>         location / {
>                 proxy_pass https://lux.1.com;
>         }
> }
> server {
>         server_name lux.1.com;
>         listen 443 ssl http2 ;
>         access_log /var/log/nginx/access.log vhost;
>         return 500;
>         ssl_certificate /etc/nginx/certs/default.crt;
>         ssl_certificate_key /etc/nginx/certs/default.key;
> }
> bash-5.0#

Any points or ideas very gratefully received. I am so sure that I must be missing something simple, but just cannot see it :(

**Comment by rr4444** [comment]
Edited the above for clarity. 

Many thanks again

**Comment by rr4444** [comment]
It hasn't changed anything except for avoiding letsencrypt limits, but i stopped using the normal LE endpoint and added the env var to use the staging server, which i see changed ok :)

**Comment by rr4444** [comment]
I've been tearing this down and redoing it many times. It's seems that the companion container just isn't making changes for each vhost in the nginx.conf and I am lost about why, especially when it manages it for one vhost. I am thinking of setting the nginx.conf manually based on the other entries so that the acme-challenge entries can exist to help the companion code, because I am giving up on this part of the automation, but it kind of loses the whole point of it. 

Very frustrating. And even if I get it going, and there is some weird behaviour, I am not sure that I can trust to it to keep going... :((

**Comment by rr4444** [comment]
BTW http on 80 works find before installing the companion container to deal with SSL certs

**Comment by buchdag** [comment]
>  It's seems that the companion container just isn't making changes for each vhost in the nginx.conf 

Just to clarify, this is **NOT** the responsibility of the companion but of `nginx-proxy`.

The companion never ever touches the `/etc/nginx/conf.d/default.conf` file.

**Comment by buchdag** [comment]
```
upstream lux.1.com {
                        # Cannot connect to network of this container
                        server 127.0.0.1 down;
                        ## Can be connected with "nginx-proxy" network
                        # wp_lux
                        server 172.18.0.4:8443;
}
```

You should not have both server directives, they are both mutually exclusive in [`nginx.tmpl`](https://github.com/nginx-proxy/nginx-proxy/blob/master/nginx.tmpl#L185-L201). I've never ever seen that happening on my setups.

This might be an `nginx-proxy` issue rather than an issue with the companion.

**Comment by rr4444** [comment]
Hi there, many thanks. I hope all is well. Thank you :)

I had answered about the env vars, but I might have misunderstood what you wanted:

```
APACHE_HTTPS_PORT_NUMBER | 8443
APACHE_HTTP_PORT_NUMBER | 8080
WORDPRESS_HTTPS_PORT | 8443
WORDPRESS_HTTP_PORT | 8080
```
are the env vars for both the working and the not working WP containers. Or do you mean do those ev's should show in a log that I have not looked at? Nothing comes up in the standard container logs on stout about ev's

The stout log has for the problem containers:

```
2020-06-04 21:24:17,301:ERROR:simp_le:1417: CA marked some of the authorizations as invalid, which likely means it could not access http://example.com/.well-known/acme-challenge/X. Did you set correct path in -d example.com:path or --default_root? Are all your domains accessible from the internet? Please check your domains' DNS entries, your host's network/firewall setup and your webserver config. If a domain's DNS entry has both A and AAAA fields set up, some CAs such as Let's Encrypt will perform the challenge validation over IPv6. If your DNS provider does not answer correctly to CAA records request, Let's Encrypt won't issue a certificate for your domain (see https://letsencrypt.org/docs/caa/). Failing authorizations: https://acme-v02.api.letsencrypt.org/acme/authz-v3/5022185976

Challenge validation has failed, see error log.
```

Thanks for clarifying re the /etc/nginx/conf.d/default.conf generated by nginxproxy, presumably  based on the shared vhost files (should have noticed that /etc/nginx/conf.d/default.conf is not on a shared volume).

**Comment by rr4444** [comment]
> ```
> upstream lux.1.com {
>                         # Cannot connect to network of this container
>                         server 127.0.0.1 down;
>                         ## Can be connected with "nginx-proxy" network
>                         # wp_lux
>                         server 172.18.0.4:8443;
> }
> ```
> 
> You should not have both server directives, they are both mutually exclusive in [`nginx.tmpl`](https://github.com/nginx-proxy/nginx-proxy/blob/master/nginx.tmpl#L185-L201). I've never ever seen that happening on my setups.
> 
> This might be an `nginx-proxy` issue rather than an issue with the companion.

Oh no...

**Comment by rr4444** [comment]
On the other hand, the working vhost has two server directives too, one 127 localhost. 

(I am new to nginx)

**Comment by rr4444** [comment]
Important note - I am setting up both nginx and the companion at the command line, but managing via Portainer. 

I am now wondering if that is causing an issue somehow. Will google

**Comment by rr4444** [comment]
FYI I just replaced the alpine version of the nginx-proxy with the 'normal' one, deleting the volumes for both that and the companion. Reinstalled both. same problem. Totally lost

**Comment by buchdag** [comment]
Could you share the commands you are using to spawn your Wordpress containers (both working and non working) ?

**Comment by rr4444** [comment]
Yes, please, thank you. 

I used to Portainer to set them up, so just used rekcod to reverse engineer the docker run commands.

I am convinced that I will kick myself once I see what is wrong, but I just can't see what I've done wrong.

The problem one:

`docker run --name wp_ls --runtime runc -v wordpress-wp_ls:/bitnami/wordpress -p 9002:8443/tcp --net wp-wp_ls --restart always -h 3f6a5d5f5c67 --expose 443/tcp --expose 8080/tcp --expose 8443/tcp -l maintainer='Bitnami <containers@bitnami.com>' -e 'PATH=/opt/bitnami/php/bin:/opt/bitnami/php/sbin:/opt/bitnami/apache/bin:/opt/bitnami/wp-cli/bin:/opt/bitnami/mysql/bin:/opt/bitnami/common/bin:/opt/bitnami/nami/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin' -e 'BITNAMI_PKG_CHMOD=-R g+rwX' -e 'HOME=/' -e 'ALLOW_EMPTY_PASSWORD=no' -e 'APACHE_ENABLE_CUSTOM_PORTS=no' -e 'APACHE_HTTPS_PORT_NUMBER=8443' -e 'APACHE_HTTP_PORT_NUMBER=8080' -e 'BITNAMI_APP_NAME=wordpress' -e 'BITNAMI_IMAGE_VERSION=5.4.1-debian-10-r6' -e 'MARIADB_HOST=mariadb' -e 'MARIADB_PORT_NUMBER=3306' -e 'MARIADB_ROOT_USER=root' -e 'MYSQL_CLIENT_CREATE_DATABASE_PRIVILEGES=ALL' -e 'MYSQL_CLIENT_ENABLE_SSL=no' -e 'NAMI_PREFIX=/.nami' -e 'OS_ARCH=amd64' -e 'OS_FLAVOUR=debian-10' -e 'OS_NAME=linux' -e 'PHP_MEMORY_LIMIT=256M' -e 'WORDPRESS_BLOG_NAME=User'\''s Blog!' -e 'WORDPRESS_DATABASE_NAME=wp_ls' -e 'WORDPRESS_DATABASE_PASSWORD=no' -e 'WORDPRESS_DATABASE_USER=wp_ls' -e 'WORDPRESS_EMAIL=user@example.com' -e 'WORDPRESS_FIRST_NAME=FirstName' -e 'WORDPRESS_HTACCESS_OVERRIDE_NONE=yes' -e 'WORDPRESS_HTACCESS_PERSISTENCE_ENABLED=yes' -e 'WORDPRESS_HTTPS_PORT=8443' -e 'WORDPRESS_HTTP_PORT=8080' -e 'WORDPRESS_LAST_NAME=LastName' -e 'WORDPRESS_PASSWORD=no' -e 'WORDPRESS_SCHEME=http' -e 'WORDPRESS_SKIP_INSTALL=no' -e 'WORDPRESS_TABLE_PREFIX=wp_' -e 'WORDPRESS_USERNAME=user' -e 'VIRTUAL_HOST=l.r.com' -e 'VIRTUAL_PORT=8443' -e 'VIRTUAL_PROTO=https' -e 'LETSENCRYPT_HOST=l.r.com' -e 'MARIADB_ROOT_PASSWORD=' -e 'MYSQL_CLIENT_CREATE_DATABASE_NAME=' -e 'MYSQL_CLIENT_CREATE_DATABASE_PASSWORD=' -e 'MYSQL_CLIENT_CREATE_DATABASE_USER=' -e 'MYSQL_CLIENT_SSL_CA_FILE=' -e 'SMTP_HOST=' -e 'SMTP_PASSWORD=' -e 'SMTP_PORT=' -e 'SMTP_PROTOCOL=' -e 'SMTP_USER=' -e 'SMTP_USERNAME=' -e 'WORDPRESS_DATABASE_SSL_CA_FILE=' -e 'WORDPRESS_EXTRA_WP_CONFIG_CONTENT=' -e 'WORDPRESS_RESET_DATA_PERMISSIONS=no' -d --entrypoint "/app-entrypoint.sh" bitnami/wordpress:latest 'httpd' '-f' '/opt/bitnami/apache/conf/httpd.conf' '-DFOREGROUND'`

The working one:

`docker run --name wp_ema --runtime runc -v wordpress-ema:/bitnami/wordpress --net wordpress-ema --restart always -h 3f6a5d5f5c67 --expose 443/tcp --expose 8080/tcp --expose 8443/tcp -l maintainer='Bitnami <containers@bitnami.com>' -e 'PATH=/opt/bitnami/php/bin:/opt/bitnami/php/sbin:/opt/bitnami/apache/bin:/opt/bitnami/wp-cli/bin:/opt/bitnami/mysql/bin:/opt/bitnami/common/bin:/opt/bitnami/nami/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin' -e 'BITNAMI_PKG_CHMOD=-R g+rwX' -e 'HOME=/' -e 'ALLOW_EMPTY_PASSWORD=no' -e 'APACHE_ENABLE_CUSTOM_PORTS=no' -e 'APACHE_HTTPS_PORT_NUMBER=8443' -e 'APACHE_HTTP_PORT_NUMBER=8080' -e 'BITNAMI_APP_NAME=wordpress' -e 'BITNAMI_IMAGE_VERSION=5.4.1-debian-10-r6' -e 'MARIADB_HOST=mariadb' -e 'MARIADB_PORT_NUMBER=3306' -e 'MARIADB_ROOT_USER=root' -e 'MYSQL_CLIENT_CREATE_DATABASE_PRIVILEGES=ALL' -e 'MYSQL_CLIENT_ENABLE_SSL=no' -e 'NAMI_PREFIX=/.nami' -e 'OS_ARCH=amd64' -e 'OS_FLAVOUR=debian-10' -e 'OS_NAME=linux' -e 'PHP_MEMORY_LIMIT=256M' -e 'WORDPRESS_BLOG_NAME=User'\''s Blog!' -e 'WORDPRESS_DATABASE_NAME=wp_ema' -e 'WORDPRESS_DATABASE_PASSWORD=no' -e 'WORDPRESS_DATABASE_USER=wp_ema' -e 'WORDPRESS_EMAIL=user@example.com' -e 'WORDPRESS_FIRST_NAME=FirstName' -e 'WORDPRESS_HTACCESS_OVERRIDE_NONE=yes' -e 'WORDPRESS_HTACCESS_PERSISTENCE_ENABLED=yes' -e 'WORDPRESS_HTTPS_PORT=8443' -e 'WORDPRESS_HTTP_PORT=8080' -e 'WORDPRESS_LAST_NAME=LastName' -e 'WORDPRESS_PASSWORD=no' -e 'WORDPRESS_SCHEME=http' -e 'WORDPRESS_SKIP_INSTALL=no' -e 'WORDPRESS_TABLE_PREFIX=wp_' -e 'WORDPRESS_USERNAME=user' -e 'VIRTUAL_HOST=dev.ema.com' -e 'VIRTUAL_PORT=8443' -e 'VIRTUAL_PROTO=https' -e 'LETSENCRYPT_HOST=dev.ema.com' -e 'MARIADB_ROOT_PASSWORD=' -e 'MYSQL_CLIENT_CREATE_DATABASE_NAME=' -e 'MYSQL_CLIENT_CREATE_DATABASE_PASSWORD=' -e 'MYSQL_CLIENT_CREATE_DATABASE_USER=' -e 'MYSQL_CLIENT_SSL_CA_FILE=' -e 'SMTP_HOST=' -e 'SMTP_PASSWORD=' -e 'SMTP_PORT=' -e 'SMTP_PROTOCOL=' -e 'SMTP_USER=' -e 'SMTP_USERNAME=' -e 'WORDPRESS_DATABASE_SSL_CA_FILE=' -e 'WORDPRESS_EXTRA_WP_CONFIG_CONTENT=' -d --entrypoint "/app-entrypoint.sh" bitnami/wordpress:latest 'httpd' '-f' '/opt/bitnami/apache/conf/httpd.conf' '-DFOREGROUND'
`

**Comment by buchdag** [comment]
Your containers have to be on the same Docker network as the nginx-proxy container.

**Comment by rr4444** [comment]
They are normally, so it's not that, but I am beginning to wonder if it is something to do with accessing the docker socket file on the host, cos portainer has to as well... 

Maybe it is not that, but I have to get this done so I am experimenting with other options so I can get something, anything going :( 

Many thanks though in the meantime :)

**Comment by buchdag** [comment]
> They are normally

Well from those command line, they're not 🤷‍♂️ 

I took the liberty to edit your previous post as that was barely readable.

Could you try running your containers with

`-e 'VIRTUAL_PORT=8443' -e 'VIRTUAL_PROTO=https' `

replaced by

`-e 'VIRTUAL_PORT=8080' `

?

**Comment by rr4444** [comment]
Agreed, regarding the command line that I shared. My mistake, as I was going totally crossyeyed with this. But I know that wasn't a problem before and must have been something to do with the rekcod command line reverse engineering. 

The 8433 change to 8000 shouldn't make a difference as both are the same WP image and one works on 8443, but honestly I have had to give up and move on. Pandemic plus this means I have a horrific time deficit. 

Many thanks for your development and efforts with me. I do appreciated it (empty words)

Warmest regards and best wishes
